### ISSUE TITLE
Wrong location loss normalizer in RPN 

### ISSUE THREAD

**Issue by zhitaoChen** [facebookresearch/detectron2#2744]
## Instructions To Reproduce the 🐛 Bug:
In detectron2/detectron2/modeling/proposal_generator/rpn.py:

```
        num_images = len(gt_labels)
        gt_labels = torch.stack(gt_labels)  # (N, sum(Hi*Wi*Ai))

        # Log the number of positive/negative anchors per-image that's used in training
        pos_mask = gt_labels == 1
        num_pos_anchors = pos_mask.sum().item()
        num_neg_anchors = (gt_labels == 0).sum().item()
        storage = get_event_storage()
        storage.put_scalar("rpn/num_pos_anchors", num_pos_anchors / num_images)
        storage.put_scalar("rpn/num_neg_anchors", num_neg_anchors / num_images)

        localization_loss = _dense_box_regression_loss(
            anchors,
            self.box2box_transform,
            pred_anchor_deltas,
            gt_boxes,
            pos_mask,
            box_reg_loss_type=self.box_reg_loss_type,
            smooth_l1_beta=self.smooth_l1_beta,
        )

        valid_mask = gt_labels >= 0
        objectness_loss = F.binary_cross_entropy_with_logits(
            cat(pred_objectness_logits, dim=1)[valid_mask],
            gt_labels[valid_mask].to(torch.float32),
            reduction="sum",
        )
        normalizer = self.batch_size_per_image * num_images
        losses = {
            "loss_rpn_cls": objectness_loss / normalizer,
            "loss_rpn_loc": localization_loss / normalizer,
        }
        losses = {k: v * self.loss_weight.get(k, 1.0) for k, v in losses.items()}
        return losses
```
which rpn location loss(i.e. "loss_rpn_loc") norm by unreasonable number rather than total positive samples. Instead, we should change it to:

```
        normalizer = self.batch_size_per_image * num_images
        losses = {
            "loss_rpn_cls": objectness_loss / normalizer,
            "loss_rpn_loc": localization_loss / torch.sum(pos_mask),
        }
        losses = {k: v * self.loss_weight.get(k, 1.0) for k, v in losses.items()}
        return losses
```

## Expected behavior:
The RPN location loss will norm by total number of positive samples, which follow by [3.1.2 in Faster-RCNN ](https://arxiv.org/abs/1506.01497) 

## Environment:
N/A, since It is a algorithmic bug

**Comment by ppwwyyxx** [comment]
This is expected. The paper uses total #positive samples divided by 10, so that "thus both cls and reg terms are roughly equally weighted". We just let them equally weighted directly.

This is how RPN was implemented in many projects.
